### PR TITLE
feat: CRUD api implemented

### PR DESCRIPTION
**Pull request by mcksyrek**

### CODE REVIEW

File: src/main/java/com/bookingplatform/server/offer/OfferService.java [hunk 1]
```diff
@@ -17,7 +19,37 @@
         return offers;
     }
 
+    public Offer getOfferById(long id) {
+        Optional<Offer> selectedOffer = offerRepository.findById(id);
+        if (selectedOffer.isPresent()) {
+            return selectedOffer.get();
+        } else {
+            throw new ResourceNotFoundException("offer");
```
[the remark this answers is not in the document]
Comment (ArtemLunt) [reply src/main/java/com/bookingplatform/server/offer/OfferService.java:27]: Also, as i know, we can throw the http error with no any magic:
https://www.baeldung.com/spring-response-status-exception#:~:text=ResponseStatus&text=This%20annotation%20was%20introduced%20in,code%20to%20an%20HTTP%20response.&text=If%20this%20exception%20is%20thrown,status%20specified%20in%20this%20annotation.

**Review by ArtemLunt — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 spring:
   jpa:
     hibernate:
-      ddl-auto: create
+      ddl-auto: none
```
[the remark this answers is not in the document]
Comment (mcksyrek) [reply src/main/resources/application.yml:4]: currently I am working on layout and that is irritating that after every reset of BE I have to create mock data again and again. any other ideas how to resolve that issue?

[the remark this answers is not in the document]
Comment (mcksyrek) [reply src/main/resources/application.yml:4]: ahh, ok got it - `none` will not make any changes if I change `@Entity`
changed to `update`

[the remark this answers is not in the document]
Comment (ArtemLunt) [reply src/main/resources/application.yml]: Exactly )

File: src/main/java/com/bookingplatform/server/offer/OfferService.java [hunk 3]
```diff
@@ -17,7 +21,31 @@
...
+            selectedOffer
+                    .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, ExceptionConstants.OFFER_NOT_FOUND));
+            offerRepository.deleteById(id);
+    }
+
+    public Offer updateOffer(long id, Offer offer) {
+        Optional<Offer> selectedOffer = offerRepository.findById(id);
+            Offer updatedOffer = selectedOffer
+                    .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, ExceptionConstants.OFFER_NOT_FOUND));
+            updatedOffer.setName(offer.getName());
```
Comment (ArtemLunt) [inline src/main/java/com/bookingplatform/server/offer/OfferService.java:46]: Doesn't seems to be extensible.
I guess we can simply save original object.

**Review by ArtemLunt — CHANGES_REQUESTED** [review]

**Review by ArtemLunt — APPROVED** [review]

**Pull request merged**